integration_tests/src/main/python/data_gen.py

```diff
@@ -685,6 +697,8 @@ def gen_scalars_for_sql(data_gen, count, seed=0, force_no_nulls=False):
 # a selection of generators that should be orderable (sortable and compareable)
 orderable_gens = [byte_gen, short_gen, int_gen, long_gen, float_gen, double_gen,
         string_gen, boolean_gen, date_gen, timestamp_gen]
+orderable_gens_with_decimal_gen = [byte_gen, short_gen, int_gen, long_gen, float_gen, double_gen,
```

Comment:
nit:  It is a lot cleaner to do

```
orderable_gens_with_decimal_gen = orderable_gens + [decimal_gen]
```

Ideally I'd like to see a comment that `orderable_gens_with_decimal_gen` would go away once all we can add decimal types to `orderable_gens` directly.

Reply:
Removed this as all support for all operators which take in `orderable_gens` is done. Have added for `eq_gens` as there is no decimal type support for `EqualNullSafe`